asm-2/cone.asm

```diff
@@ -0,0 +1,35 @@
+; --------------------------------------------------------------------------
+; Write a function that calculates the volume of a cone, given its radius
+; and height as the first two arguments.
+; 
+; Note that both arguments are given as floating points, and the return
+; value is expected to be a floating point.  This is after all an exercise
+; on floating points.
+;
+; For the value of pi, feel free to hardcode a resonable approximation
+; somewhere in your program.
+; --------------------------------------------------------------------------
+; --------------------------------------------------------------------------
+; The formula for a cone is pi * r**2 * h / 3.
+; We have r in xmm0 and h in xmm1;  We can do some math on floats
+; (section 3.11.3) and encode a constant for pi (section 3.11.4) to implement
+; this program.
+; --------------------------------------------------------------------------
+
+	default rel
+	global volume
+	section .text
+volume:
+	mulss		xmm0, xmm0		; r**2
+	mulss		xmm0, xmm1		; r**2 * h
+	mulss		xmm0, [pi]		; r ** 2 * h * pi
+	; build 3 as a float
+	mov		esi, 3			; add to integer register
```

Comment:
Two other options you could've used here are:

- Declare another constant called `three`, similar to `pi`
- Just precompute the value of `pi / 3` and use that as the one constant you declare